source/adios2/common/ADIOSTypes.inl

```diff
@@ -79,6 +79,16 @@ struct TypeInfo
     using ValueType = T;
 };
 
+// Hack "char" type into this convoluted struct definition which is the key
+// to translate between bindings API types to supported stdtypes
+template <>
+struct TypeInfo<char,
+                typename std::enable_if<std::is_integral<char>::value>::type>
```

Comment:
I think you want to use `is_same` here instead of `is_integral`:
```
template<typename T>
struct TypeInfo<T, typename std::enable_if<std::is_same<T, char>::type>
{
  using IOType = T;
  using ValueType = T;
};

Reply:
Your example leads to:

```
/home/adios/ADIOS2/bindings/CXX11/adios2/cxx11/Variable.h: In instantiation of ‘class adios2::Variable<char>’:
/home/adios/ADIOS2/bindings/CXX11/adios2/cxx11/Engine.h:453:1:   required from here
/home/adios/ADIOS2/bindings/CXX11/adios2/cxx11/Variable.h:130:11: error: ambiguous template instantiation for ‘struct adios2::TypeInfo<char, void>’
  130 |     using IOType = typename TypeInfo<T>::IOType;
```

because there is another definition that matches the T=char type. But this silly declaration works fine:

template <>
struct TypeInfo<char,
                typename std::enable_if<std::is_same<char, char>::value>::type>
{
    using IOType = char;
    using ValueType = char;
};